scripts/abins/io.py

```diff
@@ -453,27 +453,26 @@ def load(self, list_of_attributes=None, list_of_datasets=None):
         return results
 
     @staticmethod
-    def _calculate_hash(filename=None, coding='utf-8'):
+    def _calculate_hash(filename=None):
         """
         Calculates hash  of a file defined by filename according to sha512 algorithm.
 
         :param filename: name of a file to calculate hash
         :type filename: str
-        :param coding: Text encoding
-        :type encoding: str
 
         :returns: string representation of hash
 
         """
         hash_calculator = hashlib.sha512()
 
         # chop content of a file into chunks to minimize memory consumption for hash creation
-        with io.open(file=filename, mode="rt", encoding=coding, buffering=BUF, newline=None) as f:
+        with io.open(file=filename, mode="rb",
+                     buffering=BUF, newline=None) as f:
```

Comment:
I usually prefer `fd` ;-) `with open ... as f:` is idiomatic Python for context handlers over files. The ASE project however has recently standardised on `fd` (file descriptor) as it is infinitely more "greppable" while still being concise and recognisable.

I don't think there's much consistency around this in Mantid: happy to go with something more verbose if it would help. FWIW this variable is only used once, two lines down :smile:

Reply:
I like it to be obvious from the name what something is supposed to be. The other thing I dont like is that single letters variables are prone to typos

Reply:
done